runconfig/hostconfig.go

```diff
@@ -176,6 +177,7 @@ func ContainerHostConfigFromJob(job *engine.Job) *HostConfig {
 		MemorySwap:      job.GetenvInt64("MemorySwap"),
 		CpuShares:       job.GetenvInt64("CpuShares"),
 		CpusetCpus:      job.Getenv("CpusetCpus"),
+		OomKillDisable:  job.GetenvBool("OomKillDisable"),
```

Comment:
You got me wrong, normally `ContainerHostConfigFromJob` should be as simple as just return `job.GetenvJson("HostConfig", &hostConfig)`, all other codes are for backward compatibility, because we didn't have `hostConfig` at the first time.
For the new fields put into `hostConfig`, users should only use them through `hostConfig`, so if users want to use it, there would always have `HostConfig` env, and we can get the field from it, so you don't need to `job.GetenvBool("OomKillDisable")`.